File: files-objc/MediaViewController.m
// MediaViewController.m
#import "MediaViewController.h"
#import <AVKit/AVKit.h>

@interface MediaViewController ()
@property (nonatomic, strong) AVPlayerViewController *playerViewController;
@end

@implementation MediaViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupUI];
    [self loadMedia];
}

- (void)setupUI {
    self.playerViewController = [[AVPlayerViewController alloc] init];
    [self addChildViewController:self.playerViewController];
    [self.view addSubview:self.playerViewController.view];
    self.playerViewController.view.frame = self.view.bounds;
    [self.playerViewController didMoveToParentViewController:self];
    
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"返回" style:UIBarButtonItemStylePlain target:self action:@selector(goBack)];
}

- (void)loadMedia {
    NSURL *fileURL = [NSURL fileURLWithPath:self.filePath];
    AVPlayer *player = [AVPlayer playerWithURL:fileURL];
    self.playerViewController.player = player;
    [player play];
}

- (void)goBack {
    [self.navigationController popViewControllerAnimated:YES];
}

- (void)showErrorAlert:(NSString *)message {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"错误" message:message preferredStyle:UIAlertControllerStyleAlert];
    [alert addAction:[UIAlertAction actionWithTitle:@"确定" style:UIAlertActionStyleDefault handler:nil]];
    [self presentViewController:alert animated:YES completion:nil];
}

@end

File: files-swift/MediaViewController.swift
import UIKit
import AVKit

class MediaViewController: UIViewController {
    var filePath: String!
    var playerViewController: AVPlayerViewController!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setupUI()
        loadMedia()
    }
    
    private func setupUI() {
        playerViewController = AVPlayerViewController()
        addChild(playerViewController)
        view.addSubview(playerViewController.view)
        playerViewController.view.frame = view.bounds
        playerViewController.didMove(toParent: self)
        
        navigationItem.leftBarButtonItem = UIBarButtonItem(title: "返回", style: .plain, target: self, action: #selector(goBack))
    }
    
    private func loadMedia() {
        let fileURL = URL(fileURLWithPath: filePath)
        let player = AVPlayer(url: fileURL)
        playerViewController.player = player
        player.play()
    }
    
    @objc private func goBack() {
        navigationController?.popViewController(animated: true)
    }
    
    private func showErrorAlert(message: String) {
        let alert = UIAlertController(title: "错误", message: message, preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "确定", style: .default))
        present(alert, animated: true)
    }
}